tests/nosetests/pyanaconda_tests/modules/storage/module_storage_checker_test.py

```diff
@@ -108,3 +115,117 @@ def lvm_verification_test(self):
             "You must select or de-select all these disks as a set."
             .format("PHYSDISK-2", "VOLGROUP-B")
         )
+
+    def get_opal_kernel_version_test(self):
+        """Test the function for getting the firmware kernel version."""
+        patch_open = partial(patch, 'pyanaconda.modules.storage.checker.utils.open')
+
+        with patch_open() as m:
+            m.side_effect = OSError("Error!")
+            self.assertEqual(_get_opal_firmware_kernel_version(), None)
+
+        with patch_open(mock_open(read_data=" 5.10.50\n")):
+            self.assertEqual(_get_opal_firmware_kernel_version(), "5.10.50")
+
+        with patch_open(mock_open(read_data="5.10.50-openpower1-p59fd803")):
+            self.assertEqual(_get_opal_firmware_kernel_version(), "5.10.50-openpower1-p59fd803")
+
+        with patch_open(mock_open(read_data="v4.15.9-openpower1-p9e03417")):
+            self.assertEqual(_get_opal_firmware_kernel_version(), "4.15.9-openpower1-p9e03417")
+
+    def check_opal_firmware_kernel_version_test(self):
+        """Test the function for checking the firmware kernel version."""
+        check = partial(_check_opal_firmware_kernel_version)
+
+        self.assertFalse(check(None, None))
+        self.assertFalse(check("", ""))
+        self.assertFalse(check("5.09", "5.10"))
+        self.assertFalse(check("5.9.1", "5.10"))
+        self.assertFalse(check("5.9.50-openpower1-p59fd803", "5.10"))
+        self.assertFalse(check("5.8", "5.10"))
+        self.assertFalse(check("4.0-openpower1-p59fd803", "5.10"))
+
+        self.assertTrue(check("5.10", "5.10"))
+        self.assertTrue(check("5.10.1", "5.10"))
+        self.assertTrue(check("5.10.50-openpower1-p59fd803", "5.10"))
+        self.assertTrue(check("5.11", "5.10"))
+        self.assertTrue(check("6.0-openpower1-p59fd803", "5.10"))
+
+    @patch("pyanaconda.modules.storage.checker.utils.arch")
+    def opal_verification_arch_test(self, mocked_arch):
+        """Check verify_opal_compatibility with a different arch."""
+        reporter = Mock()
+        mocked_arch.get_arch.return_value = "x86_64"
+
+        verify_opal_compatibility(None, {}, None, reporter)
+        reporter.assert_not_called()
+
+    @patch("pyanaconda.modules.storage.checker.utils.arch")
+    def opal_verification_platform_test(self, mocked_arch):
+        """Check verify_opal_compatibility with a different platform."""
+        reporter = Mock()
+        mocked_arch.get_arch.return_value = "ppc64le"
+        mocked_arch.is_powernv.return_value = False
+
+        verify_opal_compatibility(None, {}, None, reporter)
+        reporter.assert_not_called()
+
+    @patch("pyanaconda.modules.storage.checker.utils._get_opal_firmware_kernel_version")
+    @patch("pyanaconda.modules.storage.checker.utils.arch")
+    def opal_verification_version_test(self, mocked_arch, version_getter):
+        """Check verify_opal_compatibility with a higher firmware version."""
```

Comment:
I don't think the test name matches the docstring.

Reply:
Fixed.